manifests/config.pp

```diff
@@ -0,0 +1,190 @@
+#
+# @param modulepath
+#   The module path for loading tasks and plan code, formatted as a string containing a list
+#   of directories. The first directory listed will be the default for downloaded modules.
+#   By default, in Bolt, this is "modules:site-modules:site" within the Bolt project directory 
+#   in `~/.puppetlabs/bolt`.
+#
+# @param color
+#   Whether to use colored output when printing messages to the console. By default, in Bolt, 
+#   this is true.
+#
+# @param concurrency
+#   The number of threads to use when executing on remote nodes. By default, in Bolt, this 
+#   is 100.
+#
+# @param format
+#   The format to use when printing results; either human or json. By default, in Bolt, this
+#   is human.
+#
+# @param hiera_config
+#   The path to the Hiera config file. By default, in Bolt, this is `hiera.yaml` in the Bolt
+#   project directory at `~/.puppetlabs/bolt`.
+#
+# @param inventoryfile
+#   Path to a structured data inventory file used to refer to groups of nodes on the command
+#   line and from plans. By default, in Bolt, this is `inventory.yaml` in the Bolt project
+#   directory at `~/.puppetlabs/bolt`.
+#
+# @param transport
+#   The default transport to use when not specified in the URL or inventory. Valid options
+#   for transport are docker, local, pcp, ssh, and winrm. By default in Bolt this is ssh.
+#
+# @param ssh_host_key_check
+#   Whether to perform host key validation when connecting over SSH. By default in Bolt this
+#   is true.
+#
+# @param ssh_password
+#   Login password for the remote system. Saving the password in plaintext is not recommended.
+#
+# @param ssh_port
+#   Connection port. By default in Bolt this is 22.
+#
+# @param ssh_private_key
+#   The path to the private key file to use for SSH authentication.
+#
+# @param ssh_proxyjump
+#   A jump host to proxy SSH connections through.
+#
+# @param ssh_tmpdir
+#   The directory to upload and execute temporarty files on the target system. The default
+#   is /var/local.
+#
+# @param ssh_user
+#   The account, used by Bolt, on the remote system. The default is the $username account
+#   specified in the user.pp manifest.
+#
+# @param ssh_run_as
+#   A different user to run commands as after login.
+#
+# @param log_console_level
+#   The type of information to display on the console. Valid options are debug, info, notice,
+#   warn, and error. The default is info.
+#
+# @param log_file
+#   The path and name of the log file.
+#
+# @param log_file_level
+#   The type of information to record in the log file. Valid options are debug, info, notice,
+#   warn, and error. By default in Bolt this is info.
+#
+# @param log_file_append
+#   Add output to an existing log file. By default in Bolt this is true.
+#
+# @param disable_analytics
+#   A different user to run commands as after login.
+#
+class simp_bolt::config (
+  String                                               $local_user         = $simp_bolt::user::username,
```

Comment:
It is possible that users may want to have all of these files somewhere for reference but do not want to actually install a user on their system.

I would probably remove these and change the behavior to have an option to include the user class. If the user class is included, then you can pick up the items from that class directly. Otherwise, you should probably just have the directory be owned by `root` and open for reading by everyone (unless there is something sensitive in there).

Reply:
Changed behavior to ensure files are created for a specific user at a specific location if both are specified; if not, will create a directory owned by root in /var/local with world read for users to use as an example.